### PR TITLE
DRILL-6280: Cleanup execution of BuildTimeScan during maven build

### PR DESCRIPTION
**Pull request by vrozov**

### CODE REVIEW

**Comment by vrozov** [comment]
@vdiravka Please review

File: common/pom.xml [hunk 1]
```diff
@@ -113,20 +113,10 @@
...
-            </goals>
-          </execution>
-        </executions>
         <configuration>
-          <mainClass>org.apache.drill.common.scanner.BuildTimeScan</mainClass>
-          <arguments>
-            <argument>${project.build.outputDirectory}</argument>
-          </arguments>
+          <additionalClasspathElements>
+            <additionalClasspathElement>${project.basedir}/src/test/resources/</additionalClasspathElement>
```
Comment (vdiravka) [inline common/pom.xml]: How it was in classpath earlier?

File: exec/java-exec/pom.xml [hunk 2]
```diff
@@ -828,31 +828,9 @@
...
       </plugin>
-      <plugin> <!-- classpath scanning  -->
+      <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>exec-maven-plugin</artifactId>
-        <version>1.2.1</version>
-        <dependencies>
-          <dependency>
-            <groupId>org.apache.drill</groupId>
-            <artifactId>drill-common</artifactId>
```
Comment (vdiravka) [inline exec/java-exec/pom.xml]: We don't need dependency onto `drill-common` or it is specified in some other place?

File: exec/java-exec/pom.xml [hunk 3]
```diff
@@ -828,31 +828,9 @@
...
-        </dependencies>
-        <executions>
-          <execution>
-            <phase>process-classes</phase>
-            <goals><goal>java</goal></goals>
-          </execution>
-        </executions>
-        <configuration>
-          <mainClass>org.apache.drill.common.scanner.BuildTimeScan</mainClass>
-          <includePluginDependencies>true</includePluginDependencies>
```
[the remark this answers is not in the document]
Comment (vrozov) [reply exec/java-exec/pom.xml]: The plugin is now configured in the `PluginManagement` section in the drill root pom.

File: common/pom.xml (hunk 1 above)
Comment (vrozov) [reply common/pom.xml]: The `logback-test.xml` was not in the classpath before, but it did not matter as slf4j did not bind to logback classic before anyway (it was not on classpath).

File: exec/java-exec/pom.xml (hunk 2 above)
Comment (vrozov) [reply exec/java-exec/pom.xml]: `drill-common` and `drill-common-test` are included as part of `test` scope dependencies

File: exec/java-exec/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (vdiravka) [reply exec/java-exec/pom.xml]: This is a nice move. 
I asked about `<includePluginDependencies>`, but you have already answered above, we just use project dependencies for now.

**Pull request closed without merging**